azure/spoke/main.tf

```diff
@@ -60,7 +52,7 @@ resource "azurerm_route_table" "main" {
 resource "azurerm_route" "main_default" {
   count = var.route_table_name != null ? 1 : 0
 
-  name                = "default"
+  name                = var.default_route_name != null ? var.default_route_name : "default"
```

Comment:
Instead of doing this, you could write:

`name = var.default_route_name`

Then set the variable default to `"default"`

Reply:
The only time you would really do default = null is if you want your default value to be a dynamic value rather than a static string

Reply:
changed